runtime/binding-kafka/src/main/java/io/aklivity/zilla/runtime/binding/kafka/internal/cache/KafkaCacheCursorFactory.java

```diff
@@ -962,6 +991,41 @@ public String toString()
             }
         }
 
+        private static final class EagerOr extends Or
+        {
+            private final List<KafkaFilterCondition> conditions;
+
+            private EagerOr(
+                List<KafkaFilterCondition> conditions)
+            {
+                super(conditions);
+                this.conditions = conditions;
+            }
+
+            @Override
+            public long test(
+                KafkaCacheEntryFW cacheEntry)
+            {
+                long accept = 0L;
+                for (int i = 0; i < conditions.size(); i++)
+                {
+                    final KafkaFilterCondition condition = conditions.get(i);
+                    final long result = condition.test(cacheEntry);
+                    if (result != 0L)
+                    {
+                        accept |= i + 1;
+                    }
```

Comment:
Let's rework this code so that it can simplify to `accept |= condition.test(cacheEntry);`.

That requires the condition to return the relevant bitmask for the filter index it is on, not just `0` or `1`.

Therefore, when each filter condition is created for non-Or conditions, it needs to know the filter index, so it can compute `1 << index` as bitmask to return when `.test(...)` is successful, instead of `1`.